dev_tools/modules.py

```diff
@@ -101,6 +113,66 @@ def list_modules(
     return result
 
 
+def get_version(search_dir: Path = _DEFAULT_SEARCH_DIR) -> Optional[str]:
+    """Check for all versions are the same and return that version.
+
+    Lists all the modules within `search_dir` (default the current working directory), checks that
+    all of them are the same version and returns that version. If no modules found, None is
+    returned, if more than one, ValueError is raised.
+
+    Args:
+        search_dir: the search directory for modules.
+    Returns:
+        None if no modules are found, the version number if exactly one version number is found.
+    Raises:
+        ValueError: if more than one version numbers are found.
+    """
+    try:
+        mods = list_modules(search_dir=search_dir, include_parent=True)
+    except ValueError:
+        return None
+    versions = {m.name: m.version for m in mods}
+    if len(set(versions.values())) > 1:
+        raise ValueError(f"Versions should be the same, instead: \n{versions}")
+    return list(set(versions.values()))[0]
+
+
+def replace_version(search_dir: Path = _DEFAULT_SEARCH_DIR, *, old: str, new: str):
+    """Replaces the current version number with a new version number.
+
+    Args:
+        search_dir: the search directory for modules.
+        old: the current version number.
+        new: the new version number.
+    Raises:
+        ValueError: if `old` does not match the current version, or if there is not exactly one
+            version number in the found modules.
+    """
+    version = get_version(search_dir=search_dir)
+    if version != old:
+        raise ValueError(f"{old} does not match current version: {version}")
+
+    _validate_version(new)
+
+    for m in list_modules(search_dir=search_dir, include_parent=True):
+        version_file = _find_version_file(search_dir / m.root)
+        content = version_file.read_text("UTF-8")
+        new_content = content.replace(old, new)
+        version_file.write_text(new_content)
+
+
+def _validate_version(new_version: str):
+    if not re.match(r"\d+\.\d+\.\d+(\.dev)?", new_version):
```

Comment:
Consider using regex from here: https://semver.org/

Reply:
Oh, cool, didn't know about it - however, interestingly it doesn't seem to match `1.2.3.dev`

Reply:
It does match `1.2.3-dev`...hmmm...